### PR TITLE
Drop support for older versions of gradle and ktlint

### PR DESCRIPTION
**Pull request by wakingrufus**
change default version of ktlint to 1.0.1
increase minimum supported gradle version to 7.4.1
 increase minimum supported ktlint version to 0.47.1
 support ktlint 1.0.1
remove retry plugin as it is included in enterprise plugin now
update code to utilize kotlin-dsl
fix ktlint errors
upgrade build to gradle 8 and add toolchain resolver
update reporter samples to ktlint 1.0.1

### CODE REVIEW

File: plugin/build.gradle.kts [hunk 1]
```diff
@@ -235,13 +187,13 @@ tasks.withType<Test>().configureEach {
...
-            maxFailures.set(10)
-        }
-    }
+//    retry {
+//        val isCiServer = System.getenv().containsKey("CI")
+//        if (isCiServer) {
+//            maxRetries.set(2)
+//            maxFailures.set(10)
+//        }
+//    }
```
Comment (JLLeitschuh) [inline plugin/build.gradle.kts:196]: If the plugin is bundled with the enterprise plugin, this should still work, right?

**Comment by JLLeitschuh** [comment]
Does the supported Gradle version match the version range supported by the Android plugin?

File: plugin/build.gradle.kts (hunk 1 above)
Comment (wakingrufus) [reply plugin/build.gradle.kts:196]: yes, I forgot to re-enable. I will fix

**Comment by wakingrufus** [comment]
> Does the supported Gradle version match the version range supported by the Android plugin?

We will be moving to requiring AGP 7.0 which requires Gradle 7.0, so this will be aligned.

**Review by JLLeitschuh — APPROVED** [review]
LGTM!

**Comment by JLLeitschuh** [comment]
Nice work on this!

**Pull request merged**